linfa-clustering/src/dbscan/algorithm.rs

```diff
@@ -0,0 +1,65 @@
+use crate::dbscan::hyperparameters::DBScanHyperParams;
+use ndarray::{Array1, ArrayBase, Axis, Data, Ix1, Ix2};
+
+pub fn predict(
+    hyperparameters: DBScanHyperParams,
+    observations: &ArrayBase<impl Data<Elem = f64> + Sync, Ix2>,
```

Comment:
Do we need a `Sync` bound here (and in the other functions)? At a first glance it doesn't seem to me that we are using any threading right now.

Reply:
No we don't, removed